src/ui/component/toast/toast.tsx

```diff
@@ -1,51 +1,140 @@
 import classNames from 'classnames'
-import { useEffect } from 'react'
+import type { FC } from 'react'
+import { useMemo, useCallback, useRef, useEffect, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import type { NotificationID, NotificationType } from '@/domain/notification/entity'
+import type { IconName } from '@/ui/component/icon/icon'
+import { Icon } from '@/ui/component/icon/icon'
+import { useOnClickOutside } from '@/ui/hook/useOnClickOutside'
+import { useOnKeyboard } from '@/ui/hook/useOnKeyboard'
+import { Button } from '@/ui/component//button/button'
+import type { DismissNotificationInput } from '@/domain/notification/aggregate'
+import '@/ui/component/notifications/i18n'
 import './toast.scss'
 
-type ToastVariants = 'success' | 'error' | 'warning' | 'info'
+type ToastCTAProps = {
+  type: NotificationType
+  action: string
+}
+const ToastCTA: FC<ToastCTAProps> = ({ action, type }) => {
+  const { t } = useTranslation('notifications')
+
+  const handleClick = useCallback(() => {
+    window.location.reload()
+  }, [])
+  return (
+    <Button
+      className={classNames('okp4-dataverse-portal-toast-cta', type)}
+      label={t(`action.${action}`)}
+      onClick={handleClick}
+    />
+  )
+}
```

Comment:
This component looks not extensible.

For now, it only handles refresh actions, but how will you do to manage more actions in a next future if you hard code `window.location.reload()` in  `handleClick` function?

Reply:
As is it's not extensible. We might define an object with key props as `ActionType`, and value would be the function that run side effects. What do you think?